tests/test_plugins/test_invdes.py

```diff
@@ -1,31 +1,28 @@
+# ruff: noqa: F811
 # Test the inverse design plugin
 
 import pytest
-import builtins
 
 import numpy as np
-import jax.numpy as jnp
+import autograd.numpy as anp
 
 import tidy3d as td
-import tidy3d.plugins.adjoint as tda
 import tidy3d.plugins.invdes as tdi
 
 # use single threading pipeline
-from . import test_adjoint as ta
 
 from .test_adjoint import use_emulated_run, use_emulated_run_async  # noqa: F401
 
-from ..utils import run_emulated, assert_log_level, AssertLogLevel
+from ..utils import run_emulated, run_async_emulated, AssertLogLevel
```

Comment:
Also a general comment, but if these fixtures are put into `conftest.py` instead of `utils`, then they would be available to pytest without importing. Would fix all of the `F401` warnings. We should probably restructure all of the tests a bit.

Reply:
If it's only this change then #1662 probably makes sense, I'll look into it. At a later stage we'll need to look at the tests a bit more closely too though.